test/performance/range/gap_decorator_benchmark.cpp

```diff
@@ -0,0 +1,460 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <algorithm>
+#include <array>
+#include <cstdlib>
+#include <chrono>
+#include <climits>
+#include <cmath>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+#include <random>
+#include <utility>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alignment/exception.hpp>
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/decorator/all.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/std/ranges>
+
+using namespace seqan3;
+
+typedef long double time_type;
+
+/* Helper function to sample gap length for each ungapped sequence position.
+ *
+ * Parameter:
+ * size_type    size type of underlying sequence range
+ *
+ * Arguments:
+ * gap_vector   reference to empty vector storing the sampled gap lengths
+ * size         final (aligned) sequence size
+ */
+template<typename size_type>
+void sample(std::vector<size_type> * gap_vector, size_type size)
+{
+    std::default_random_engine generator;
+    generator.seed(std::chrono::system_clock::now().time_since_epoch().count());
```

Comment:
I'd prefer to keep it random, because of code coverage. There is currently no randomization used inside the unit tests (I know performance is supposed to measure only performance). Apart from this I think the gap sampling here should not produce significant different results between runs (for larger sequence lengths). But it is a point to rethink when the unit tests are done.